### DISCUSSION TITLE
Anyway to access forwarded ports through `host:port`?

### DISCUSSION THREAD

**Discussion by F-Unction** [community/community#28528] [Codespaces] [answered]
It seems I can't access tcpserver on codespaces through an address like`https://xx
x-codespace-xxx-port.githubpreview.dev`

Anyway to make it? or to get `host:port` like address?

**Comment by jaclynj** [comment]
What have you tried so far? Are you attempting this from codespaces in the browser, from VSCode desktop extension, or from the GitHub CLI?

Have you followed the port forwarding instructions? https://docs.github.com/en/codespaces/developing-in-codespaces/forwarding-ports-in-your-codespace

**Reply by F-Unction** [reply]
I am trying from codespaces in the browser, using the way of documents.

But I got an address like `https://xx x-codespace-xxx-port.githubpreview.dev`, and I can't connect to my tcpserver using it.

**Comment by legomushroom** [comment, marked as answer]
@F-Unction nope, there is currently no way to do that with the public port URLs. It was on our roadmap a while back, but was deprioritized since we dind't have a lot of feedback regarding use cases to support something like this.

As I understand from the question, your use case is that you want to do a TCP connection directly to that public port, but since it is behind the `HTTPS` endpoint it does not work as you would expect?

The only way to do raw `TCP` right now is to use a desktop VSCode client to connect to that Codespace, which will bind all ports to `127.0.0.1` address, so you would be able to work with `127.0.0.1:port`. Then you can alias the `127.0.0.1` IP address to some host name on your machine, to get the `host:port` pair as you want, but it will be on your local machine only, of course.

**Reply by F-Unction** [reply]
Ok, thank you!

**Reply by chwan1** [reply]
@legomushroom is there any way I can prevent desktop VSCode to bind ports to 127.0.0.1? Or is there any way I can get a pubic accessible address from the desktop App? I'm working on a PWA project so I need the `https://xxx-codespace-xxx-port.githubpreview.dev` address to test on another device.

**Reply by F-Unction** [reply]
> @legomushroom is there any way I can prevent desktop VSCode to bind ports to 127.0.0.1? Or is there any way I can get a pubic accessible address from the desktop App? I'm working on a PWA project so I need the address to test on another device.`https://xxx-codespace-xxx-port.githubpreview.dev`

Turn the port into public may help
